tests/e2e/config_secret.go

```diff
@@ -0,0 +1,311 @@
+		}
+
+		ginkgo.By("Set search level permission for testuser1 and testuser2")
+		err = setSearchlevelPermission(vcAddress, sshClientConfig, masterIp, configSecretUser1Alias,
+			configSecretTestUser1)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = setSearchlevelPermission(vcAddress, sshClientConfig, masterIp, configSecretUser2Alias,
+			configSecretTestUser2)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			ginkgo.By("Delete testUser1, its roles and permissions")
+			err = deleteUsersRolesAndPermissions(vcAddress, sshClientConfig, masterIp,
+				configSecretTestUser1, configSecretUser1Alias, dataCenters, clusters, hosts, vms, datastores)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By("Delete testUser2, its roles and permissions")
+			err = deleteUsersRolesAndPermissions(vcAddress, sshClientConfig, masterIp,
+				configSecretTestUser2, configSecretUser2Alias, dataCenters, clusters, hosts, vms,
+				datastores)
+		}()
+
+		ginkgo.By("Fetch vsphere-config-secret file details")
+		secretConf = getConfigSecretFileValues(client, ctx)
+
+		ginkgo.By("Delete previously created vsphere-config-secret file")
+		err = deleteCsiVsphereConfFile(sshClientConfig, masterIp)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create vsphere-config-secret file with testuser1 credentials")
+		err = createCsiVsphereConfFile(secretConf.Global.ClusterID,
+			secretConf.Global.ClusterDistribution, secretConf.Global.VCenterHostname,
+			secretConf.Global.InsecureFlag,
+			configSecretUser1Alias, configSecretTestUser1Password, secretConf.Global.VCenterPort,
+			secretConf.Global.Datacenters, sshClientConfig, masterIp)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Restart CSI driver")
+		err = restartCSIDriver(ctx, sshClientConfig, masterIp, client, csiNamespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify we can create a PVC and attach it to pod")
+		ginkgo.By("Creating Storage Class")
+		storageclass, err := createStorageClass(client, nil, nil, "", "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			ginkgo.By("Delete Storage Class")
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating Pvc")
+		pvclaim1, err := createPVC(client, namespace, nil, "", storageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		// Waiting for PVC to be bound.
+		var pvclaims1 []*v1.PersistentVolumeClaim
+		pvclaims1 = append(pvclaims1, pvclaim1)
+		ginkgo.By("Waiting for all claims to be in bound state")
+		_, err = fpv.WaitForPVClaimBoundPhase(client, pvclaims1, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim1.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
Check if the CNS volume is deleted or not

Reply:
done